src/System.Net.WebHeaderCollection/src/System/Net/HeaderInfo.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net
+{
+    internal delegate string[] HeaderParser(string value);
```

Comment:
It'd be interesting to understand if this is really needed or whether any uses could instead be a ```Func<string, string[]>```.  Not something that needs to be changed as part of this PR, just something to consider for later.